internal/service/alarms/internal/alarms_server.go

```diff
@@ -91,11 +92,43 @@ func (a *AlarmsServer) GetProbableCause(ctx context.Context, request api.GetProb
 	return nil, fmt.Errorf("not implemented")
 }
 
+// AmNotification Used internally, alertmanager from ACM calls here.
+// Note: the errors returned can also be view under alertmanager pod logs but also logging here for convenience
 func (a *AlarmsServer) AmNotification(ctx context.Context, request api.AmNotificationRequestObject) (api.AmNotificationResponseObject, error) {
-	// TODO: Implement the logic to handle the AM notification
-	slog.Debug("Received AM notification", "groupLabels", request.Body.GroupLabels)
-	for _, alert := range request.Body.Alerts {
-		slog.Debug("Alert", "fingerprint", alert.Fingerprint, "startsAt", alert.StartsAt, "status", alert.Status)
+	// Audit the table with full list of alerts in the current payload. If missing set them to resolve
+	if err := a.AlarmsRepository.ResolveNotificationIfNotInCurrent(ctx, request.Body); err != nil {
```

Comment:
Are we sure that a single notification has all alerts? Say, alert1, alert2 and alert3 are firing and that the group_interval is 1m. If there are no changes, nothing will be reported after 1 m, but what if a new alert comes in in between. Will am wait until it completes 1m to report all of them? or does report it right away? group_interval applies when the parent receiver has no group_by?